From 11627d2393d8da721158875a8320b6ac370cfbe8 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 19 Aug 2025 13:47:27 +0100
Subject: [PATCH] config: Allow MFD_RP1 as well as MISC_RP1

While we continue to use the downstream RP1 driver, update some other
Kconfig settings to recognise MFD_RP1 as a valid RP1 driver.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/clk/Kconfig     | 2 +-
 drivers/pinctrl/Kconfig | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/clk/Kconfig
+++ b/drivers/clk/Kconfig
@@ -89,7 +89,7 @@ config COMMON_CLK_RK808
 
 config COMMON_CLK_RP1
 	tristate "Raspberry Pi RP1-based clock support"
-	depends on MISC_RP1 || COMPILE_TEST
+	depends on MISC_RP1 || MFD_RP1 || COMPILE_TEST
 	default MISC_RP1
 	help
 	  Enable common clock framework support for Raspberry Pi RP1.
--- a/drivers/pinctrl/Kconfig
+++ b/drivers/pinctrl/Kconfig
@@ -674,7 +674,7 @@ config PINCTRL_MLXBF3
 
 config PINCTRL_RP1
 	tristate "Pinctrl driver for RP1"
-	depends on MISC_RP1
+	depends on MISC_RP1 || MFD_RP1
 	default MISC_RP1
 	select PINMUX
 	select PINCONF
